src/jdk.jartool/share/classes/sun/security/tools/jarsigner/Main.java

```diff
@@ -1444,6 +1473,16 @@ String printCert(boolean isTsCert, String tab, Certificate c,
         }
 
         if (x509Cert != null) {
+            PublicKey key = x509Cert.getPublicKey();
+            String sigalg = x509Cert.getSigAlgName();
+
+            // process the certificate in the signer's cert chain to see if
+            // weak algorithms are used, and provide warnings as needed.
+            certStr.append("\n").append(tab)
+                    .append("Signature algorithm: ")
+                    .append(checkWeakAlg(sigalg))
```

Comment:
If the cert is trusted, I don't think we should print a warning if the signature algorithm is weak. Otherwise this will generate false warnings for SHA-1 roots which are not an issue.  You should check the key size though. And you can still print the signature algorithm. You may need to move line 1489-1490 before this to first determine if the cert is trusted.

Reply:
Fixed to not check the signature algorithm for a trusted cert, and updated the test accordingly.